src/overlays/actors/ovl_Boss_05/z_boss_05.c

```diff
@@ -159,107 +341,1352 @@ ActorInit Boss_05_InitVars = {
+            Math_ApproachF(&this->lowerJawScaleXZ, 1.5f, 1.0f, 0.7f);
+            this->lilyPadWithHeadMovementState = BIO_BABA_LILY_PAD_WITH_HEAD_MOVEMENT_STATE_LUNGE_ATTACK;
+
+            if (this->timers[TIMER_CURRENT_ACTION] == 0) {
+                this->lilyPadWithHeadAttackState = BIO_BABA_LILY_PAD_WITH_HEAD_ATTACK_STATE_IDLE;
+                this->timers[TIMER_CURRENT_ACTION] = 30;
+            }
+            break;
+
+        default:
+            break;
+    }
+
+    hitReaction = Boss05_LilyPadWithHead_UpdateDamage(this, play);
+    if ((hitReaction != BIO_BABA_HEAD_HIT_REACTION_NONE_OR_DAMAGED) || (this->forceDetachTimer == 1)) {
+        Boss05* child;
+
+        this->dyna.actor.params = BIO_BABA_TYPE_LILY_PAD;
+        this->actionFunc = Boss05_LilyPad_Idle;
+        this->dyna.actor.flags &= ~ACTOR_FLAG_TARGETABLE;
+        func_800BC154(play, &play->actorCtx, &this->dyna.actor, ACTORCAT_BG);
+        if (this->forceDetachTimer != 0) {
+            hitReaction = BIO_BABA_HEAD_HIT_REACTION_DEATCH;
+        }
+
+        if (hitReaction >= BIO_BABA_HEAD_HIT_REACTION_DEATCH) {
+            child = (Boss05*)Actor_SpawnAsChild(
+                &play->actorCtx, &this->dyna.actor, play, ACTOR_BOSS_05, this->dyna.actor.world.pos.x,
+                this->dyna.actor.world.pos.y, this->dyna.actor.world.pos.z, this->dyna.actor.shape.rot.x,
+                this->dyna.actor.shape.rot.y, this->dyna.actor.shape.rot.z, BIO_BABA_TYPE_FALLING_HEAD);
+
+            if (child != NULL) {
+                Player* player2 = GET_PLAYER(play);
+
+                if (player2->lockOnActor == &this->dyna.actor) {
+                    player2->lockOnActor = &child->dyna.actor;
+                    play->actorCtx.targetCtx.fairyActor = &child->dyna.actor;
+                    play->actorCtx.targetCtx.lockOnActor = &child->dyna.actor;
+                }
+
+                for (i = 0; i < BIO_DEKU_BABA_LILY_PAD_LIMB_MAX; i++) {
+                    child->lilyPadSkelAnime.jointTable[i] = this->lilyPadSkelAnime.jointTable[i];
+                }
+
+                for (i = 0; i < BIO_DEKU_BABA_HEAD_LIMB_MAX; i++) {
+                    child->headSkelAnime.jointTable[i] = this->headSkelAnime.jointTable[i];
+                }
+
+                Actor_PlaySfx(&this->dyna.actor, NA_SE_EN_MIZUBABA1_DAMAGE);
+
+                if (hitReaction > BIO_BABA_HEAD_HIT_REACTION_DEATCH) {
+                    child->drawDmgEffState = hitReaction - BIO_BABA_HEAD_HIT_REACTION_DEATCH;
+                }
+            }
+        } else if (hitReaction == BIO_BABA_HEAD_HIT_REACTION_DEAD) {
+            for (i = 0; i < 2; i++) {
+                child = (Boss05*)Actor_SpawnAsChild(&play->actorCtx, &this->dyna.actor, play, ACTOR_BOSS_05,
+                                                    this->headPos.x, this->headPos.y, this->headPos.z, this->headRot.x,
+                                                    this->headRot.y, this->headRot.z, i + BIO_BABA_TYPE_FRAGMENT_BASE);
```

Comment:
```suggestion
                                                    this->headRot.y, this->headRot.z, BIO_BABA_TYPE_FRAGMENT_BASE + i);
```
Just the tiniest nit and feel free to ignore. I find reading these as `base + offset` easier to follow than `offset + base` if that makes sense?